core/domain/acl_decorators.py

```diff
@@ -126,3 +168,238 @@ def test_can_view_stats(self, exploration_id, **kwargs):
             raise self.PageNotFoundException
 
     return test_can_view_stats
+
+
+def can_create_exploration(handler):
+    """Decorator to check whether the user can create an exploration."""
+
+    def test_can_create(self, **kwargs):
+        if self.user_id is None:
+            raise self.NotLoggedInException
+
+        if role_services.ACTION_CREATE_EXPLORATION in self.actions:
+            return handler(self, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to create an exploration.')
+
+    return test_can_create
+
+
+def can_create_collection(handler):
+    """Decorator to check whether the user can create a collection."""
+
+    def test_can_create(self, **kwargs):
+        if self.user_id is None:
+            raise self.NotLoggedInException
+
+        if role_services.ACTION_CREATE_COLLECTION in self.actions:
+            return handler(self, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to create a collection.')
+
+    return test_can_create
+
+
+def can_access_creator_dashboard(handler):
+    """Decorator to check whether the user can access creator dashboard
+    page.
+    """
+
+    def test_can_access(self, **kwargs):
+        if self.user_id is None:
+            raise self.NotLoggedInException
+
+        if role_services.ACTION_ACCESS_CREATOR_DASHBOARD in self.actions:
+            return handler(self, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to access creator dashboard.')
+
+    return test_can_access
+
+
+def can_view_exploration_feedback(handler):
+    """Decorator to check whether the user can view feedback for a given
+    exploration.
+    """
+
+    def test_can_access(self, exploration_id, **kwargs):
+        if check_exploration_editable(
+                self.user_id, self.actions, exploration_id):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to view feedbacks.')
```

Comment:
ok. 
Changed it's permission as well. As now feedback threads should be accessible to all logged in users (exploration editors).